### PR TITLE
🐛  re-fetch entitlements on reset

### PR DESCRIPTION
**Pull request by jpettitt**
re-fetch entitlements for all platforms in `reAuthorizePlatform`.  

Notable change: `reAuthorizePlatform`  is now `resetPlatforms` - calls `initializePlatformnStore_` to reset to starting condition.

### CODE REVIEW

File: extensions/amp-subscriptions/0.1/amp-subscriptions.js [hunk 1]
```diff
@@ -480,29 +480,27 @@ export class SubscriptionService {
...
+    this.subscriptionAnalytics_.serviceEvent(
+        SubscriptionAnalyticsEvents.PLATFORM_REAUTHORIZED,
+        subscriptionPlatform.getServiceId()
+    );
+    // deprecated event fired for backward compatibility
+    this.subscriptionAnalytics_.serviceEvent(
+        SubscriptionAnalyticsEvents.PLATFORM_REAUTHORIZED_DEPRECATED,
+        subscriptionPlatform.getServiceId()
+    );
+    this.platformStore_.reset();
```
Comment (dvoytenko) [inline extensions/amp-subscriptions/0.1/amp-subscriptions.js]: I think this should go as a first line. Probably not essential for the current exact code, but would be more correct as a flow.

Comment (jpettitt) [reply extensions/amp-subscriptions/0.1/amp-subscriptions.js]: Agree, done

File: extensions/amp-subscriptions/0.1/amp-subscriptions.js [hunk 2]
```diff
@@ -478,27 +478,31 @@ export class SubscriptionService {
...
-          platform.reset();
-          this.platformStore_.resetEntitlementFor(
-              platform.getServiceId());
-          this.fetchEntitlements_(platform);
-        });
+  resetPlatforms() {
+    const serviceIds = this.platformConfig_['services'].map(service =>
+      service['serviceId'] || 'local');
+
+    this.initializePlatformStore_(serviceIds);
```
Comment (prateekbh) [inline extensions/amp-subscriptions/0.1/amp-subscriptions.js]: Wouldn't this still lose the references to the platforms which have registered themselves by calling 

https://github.com/ampproject/amphtml/blob/master/extensions/amp-subscriptions/0.1/amp-subscriptions.js#L172

**Comment by prateekbh** [comment]
LGTM

**Pull request merged**